### PR TITLE
LA-222 : Download Received Shares

### PR DESCRIPTION
**Pull request by imGok**

### CODE REVIEW

**Comment by imGok** [comment]
Fixed a bug with download in My Space it was crashing. 
I didn't make multiple download yet only single 
review pls

File: domain/lib/src/usecases/received/received_share_view_state.dart [hunk 1]
```diff
@@ -85,4 +85,22 @@ class ReceivedShareContextMenuItemViewState extends ViewState {
...
+  final List<DownloadTaskId> taskIds;
+
+  DownloadReceivedShareViewState(this.taskIds);
+
+  @override
+  List<Object> get props => [taskIds];
+}
+
+class DownloadReceivedShareFailure extends FeatureFailure {
+  final Exception downloadFileException;
```
Comment (manhduc-nguyen) [inline domain/lib/src/usecases/received/received_share_view_state.dart:100]: should be `Object` or `dynamic` type

File: lib/presentation/widget/received/received_share_viewmodel.dart [hunk 2]
```diff
@@ -127,4 +130,46 @@ class ReceivedShareViewModel extends BaseViewModel {
...
+
+  void _download(List<ShareId> shareIds, {ItemSelectionType itemSelectionType = ItemSelectionType.single}) {
+    store.dispatch(_downloadFileAction(shareIds));
+    _appNavigation.popBack();
+    if (itemSelectionType == ItemSelectionType.multiple) {
+      cancelSelection();
+    }
+  }
+
+  ThunkAction<AppState> _downloadFileAction(List<ShareId> shareIds) {
```
Comment (manhduc-nguyen) [inline lib/presentation/widget/received/received_share_viewmodel.dart:168]: this should be `OnlineThunkAction`

**Review by hoangdat — COMMENTED** [review]
LGTM, go ahead with multiple download

**Comment by imGok** [comment]
@hoangdat @manhduc-nguyen up to date, but couldn't do for iOS too should I let only android for now?

**Review by manhduc-nguyen — APPROVED** [review]
LGTM

**Comment by manhduc-nguyen** [comment]
> @hoangdat @manhduc-nguyen up to date, but couldn't do for iOS too should I let only android for now?

yes, u just have to do it for Android. I''ll cover for iOS later

**Review by hoangdat — APPROVED** [review]

**Comment by huynguyennovem** [comment]
Tested on Android 8.0 & Android 11, I got the error log:
```
No permissions found in manifest: 15
```
It should display a request permission dialog for writing file into storage.
Suggest solution: Upgrade lib `permission_handler` to `5.1.0+2`:
```
permission_handler: 5.1.0+2
```

**Comment by dab246** [comment]
I test on all android  also has the same error:
`No permissions found in manifest`

**Comment by hoangdat** [comment]
@huynguyennovem please add your fix in another PR

**Pull request merged**